src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -230,21 +230,31 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                     if ((sqlConn != null) && !connInfo.IsSqlDW && !connInfo.IsAzure)
                     {
                         BackupOperation backupOperation = DisasterRecoveryService.Instance.SetBackupInput(helper.DataContainer, sqlConn, backupParams.BackupInfo);
+                        SqlTask sqlTask = null;
 
                         // create task metadata
                         TaskMetadata metadata = new TaskMetadata();
                         metadata.ServerName = connInfo.ConnectionDetails.ServerName;
                         metadata.DatabaseName = connInfo.ConnectionDetails.DatabaseName;
-                        metadata.Name = SR.Backup_TaskName;
-                        metadata.IsCancelable = true;
                         metadata.Data = backupOperation;
-                        
-                        // create backup task and perform
-                        SqlTask sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupTaskAsync);
-                        sqlTask.Run();
+                        metadata.IsCancelable = true;
+
+                        if (backupParams.IsScripting)
+                        {
+                            metadata.Name = string.Format("{0} {1}", SR.BackupTaskName, SR.ScriptTaskName);
+                            metadata.TaskExecutionMode = TaskExecutionMode.Script;
+                            sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupScriptTaskAsync);
```

Comment:
again, you should never directly refer to the .Instance of other services as it blocks all unit testing. Create a local property "TaskManagerInstance" and allow tests to override its value (so you can pass in a mock).
Ideally, use the newer service implementation that takes in a ServiceProvider so you never need to do manually override, instead tests pass in a different service provider with mocks.